### ISSUE TITLE
A inference demo 

### ISSUE THREAD

**Issue by leospark** [LMSAudio/Complex_PF#2]
Thanks for your great job. 
I'm still confused about some details, for example, liner filter and whether this method can be real-time(one frame in and one frame out.).  May you share a sample inference demo?  @ModarHalimeh

**Comment by leospark** [comment]
`   
>    nrSamples = len(item_name)
>         for sampleId in range(0, nrSamples):
>             s_hat = out_audio[sampleId, :]
>             file = item_name[sampleId]
>             sig = {}
>             sig['s_hat'] = s_hat.detach().cpu().numpy()
>             scipy.io.savemat('./results/' + file + '.mat', sig)
>             data = s_hat.detach().cpu().numpy()              # my code 
>             data = data / np.abs(data).max()
>             sf.write('./results/' + file + '.wav', data, 16000)
`
I try saving enhanced wav files during testing above, but got abnormal output wav file. 
What's wrong with the code? Or how should I save processed files?

**Comment by zuowanbushiwo** [comment]
I also look forward to a demo and a model

**Comment by ModarHalimeh** [comment]
@liziru Hello. Sorry for being a bit late with this. 
Regarding the linear filter, we unfortunately did not publish it. However, that should not matter much, you can in principle use any AEC method to remove the early part of the echo in the microphone signal. The application of the postfilter (i.e., the mask estimation) takes place after the postfilter using the residual echo signal, i.e., the microphone signal after removing the early echo part. 

Regarding the online demo: I will try to allocate some time to that and I hope I will be able to upload it here soon.

**Comment by leospark** [comment]
> @liziru Hello. Sorry for being a bit late with this. Regarding the linear filter, we unfortunately did not publish it. However, that should not matter much, you can in principle use any AEC method to remove the early part of the echo in the microphone signal. The application of the postfilter (i.e., the mask estimation) takes place after the postfilter using the residual echo signal, i.e., the microphone signal after removing the early echo part.
> 
> Regarding the online demo: I will try to allocate some time to that and I hope I will be able to upload it here soon.

Thanks!

**Comment by csu-roycheng** [comment]
> `
> 
> > nrSamples = len(item_name)
> > for sampleId in range(0, nrSamples):
> > s_hat = out_audio[sampleId, :]
> > file = item_name[sampleId]
> > sig = {}
> > sig['s_hat'] = s_hat.detach().cpu().numpy()
> > scipy.io.savemat('./results/' + file + '.mat', sig)
> > data = s_hat.detach().cpu().numpy()              # my code
> > data = data / np.abs(data).max()
> > sf.write('./results/' + file + '.wav', data, 16000)
> > `
> > I try saving enhanced wav files during testing above, but got abnormal output wav file.
> > What's wrong with the code? Or how should I save processed files?

Hi @liziru, I ran the code and got the strange wav file too.
Do you figure out the problem?

**Comment by ModarHalimeh** [comment]
Hey  @csu-roycheng. I would be grateful if you could explain what you mean by "strange" wav file? Is it possible to provide the wav files of the output, microphone and far-end signals? 
You can send them via e-mail if you like.

**Comment by csu-roycheng** [comment]
Hi, thanks for your reply. I trained your model using your dataset provided in the source_code/data folder, and the output wav file is very strange, like distortion. Then I found the dataset is too small, so I changed the dataset to AEC Challenge dataset. I trained the model again, but nothing changed.


8199_out.wav is the output wav using the trained model.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LMSAudio/Complex_PF"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月22日(星期三) 下午5:51
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [LMSAudio/Complex_PF] A inference demo  (#2)





 
Hey  @csu-roycheng. I would be grateful if you could explain what do you mean by "strange" wav file? Is it possible to provide the wav files + the microphone and far-end signals?
 You can send them via e-mail if you like.

**Comment by ModarHalimeh** [comment]
Hi, thanks for getting back to me quickly :)

As you've already noted, the dataset we provided is a dummy one (acts as a place holder) and of course it is *very* small and not sufficient for the model to learn extracting the source meaningfully. 

However, when trained using a large dataset, e.g., the AEC challenge one, I would've expected the model to perform much better. 

Unfortunately, I can not see an attached file. Maybe you can try e-mailing it to me? With also some description of what pre-processing steps you are using for the dataset.

**Comment by csu-roycheng** [comment]
Hi, thanks for your reply. I have uploaded the wav files, and 8199_out.wav is the output file using the trianed model. I use the synthetic dataset of AEC Challenge, and preprocess the training data to the hdf5 file. The structure of the hdf5 file is the same as your dummy one.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LMSAudio/Complex_PF"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月22日(星期三) 晚上7:21
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [LMSAudio/Complex_PF] A inference demo  (#2)





 
Hi, thanks for getting back to me quickly :)
 
As you've already noted, the dataset we provided is a dummy one (acts as a place holder) and of course it is very small and not sufficient for the model to learn extracting the source meaningfully.
 
However, when trained using a large dataset, e.g., the AEC challenge one, I would've expected the model to perform much better.
 
Unfortunately, I can not see an attached file. Maybe you can try e-mailing it to me? With also some description of what pre-processing steps you are using for the dataset.

**Comment by ModarHalimeh** [comment]
Thanks again for replying. Unfortunately, I can not access the files you are mentioning. Perhaps your fork or branch is set to private?
